src/services/messagesService.js

```diff
@@ -25,21 +25,36 @@ import { generateOcsUrl } from 'nextcloud-router'
 
 const CANCEL_TOKEN = axios.CancelToken
 
-/**
- * Fetches messages that belong to a particular conversation
- * specified with its token.
- *
- * @param {string} token The conversation token;
- */
-const fetchMessages = async function(token) {
-	try {
-		const response = await axios.get(generateOcsUrl('apps/spreed/api/v1/chat', 2) + token + '?lookIntoFuture=0')
-		return response
-	} catch (exception) {
-		if (!exception.response || !exception.response.status || exception.response.status !== 304) {
-			console.debug('Error while fetching messages: ', exception)
+const cancelableFetchMessages = function() {
+	/**
+	 * cancelToken= the token that gets injected into the axios method and links it
+	 * to the cancel function;
+	 * cancel= function that allows to delete the api call;
+	 */
+	const { token: cancelToken, cancel: cancelFetchMessages } = CANCEL_TOKEN.source()
+	/**
+	 * Fetches messages that belong to a particular conversation
+	 * specified with its token.
+	 *
+	 * @param {string} token The conversation token;
```

Comment:
https://github.com/nextcloud/gallery/blob/refactor/gallery-19/src/utils/CancelableRequest.js
Use this. You'll get much cleaner code on your requests.

Reply:
https://github.com/nextcloud/gallery/blob/6fa4fa6fbafef7dc597927f9b3dd2e581ddade1b/src/components/Folder.vue#L57
https://github.com/nextcloud/gallery/blob/6fa4fa6fbafef7dc597927f9b3dd2e581ddade1b/src/components/Folder.vue#L114-L128

Reply:
I am planning use that, but shouldn't it be in another pr?